slam-libraries/viam-cartographer/src/slam_service/slam_service_test.cc

```diff
@@ -69,45 +72,308 @@ BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_mapping) {
     slamService.occupied_space_weight = 9990.09;
     slamService.translation_weight = 9989.89;
     slamService.rotation_weight = 9988.88;
+
+    BOOST_TEST(slamService.GetActionMode() == SLAMServiceActionMode::MAPPING);
     slamService.OverwriteMapBuilderParameters();
     checkCartoMapBuilderParameters(slamService);
 }
 
 BOOST_AUTO_TEST_CASE(
     OverwriteMapBuilderParameters_check_default_values_mapping) {
+    // Mapping is the default action_mode when slamService is created
+    SLAMServiceImpl slamService;
+
+    BOOST_TEST(slamService.GetActionMode() == SLAMServiceActionMode::MAPPING);
+    slamService.OverwriteMapBuilderParameters();
```

Comment:
That's a valid concern and I considered it. I am sticking to using `overwrite` though because that's what the function actually does. It overwrites the map_builder parameters with the parameters we defined in `slamService`. In this case, we overwrite map_builder parameters with the default slamService parameters.